fix(helm): helm login should not use host path

rarkins opened the pull request:
<!-- If this is your first pull request: sign the CLA with this GitHub app: https://cla-assistant.io/renovatebot/renovate -->
<!-- Make sure the `Allow edits and access to secrets by maintainers` checkbox is checked on this pull request. -->
<!-- Please read https://github.com/renovatebot/renovate/blob/main/.github/contributing.md before you create your pull request.-->

## Changes

Strips path part from host for "helm login"

## Context

- #36052

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

<!-- Do you have any suggestions about this PR template? Edit it here: https://github.com/renovatebot/renovate/edit/main/.github/pull_request_template.md -->

<!-- Please do not force push to your PR's branch after you have created your PR, as doing so forces us to review the whole PR again. This makes it harder for us to review your work because we don't know what has changed. -->
<!-- PRs will always be squashed by us when we merge your work. Commit as many times as you need in this branch. -->

rarkins committed:
fix(helm): helm login should not use host path

rarkins committed:
lint-fix

secustor reviewed (commented):
I think this will break, registries on sub paths. Like Artifactory or Nexus

rarkins commented:
@secustor see https://github.com/renovatebot/renovate/discussions/36052#discussioncomment-13215886

Are there other commands which we use this for where the subpath is valid?

viceice commented:
i think this schould only be done for oci charts. then they use the docker protocoll and that requires to be `https://<host>/v2/` base url.

secustor commented:
Not that I'm aware of. 

`helm registry login` is already only for OCI images. 
"standard" helm registries use [`helm repo add`](https://helm.sh/docs/helm/helm_repo_add/)

rarkins committed:
simplify

rarkins commented:
Here's what I've now changed:
- `helm registry login` was the only way this function was called
- `helm registry login` _needs_ the path stripped
- I removed the generic `loginCMD` parameter so that now it's hardcoded to `helm registry login` and we are safe to strip path. Nobody should change this by accident to something non-functional

rarkins committed:
Merge remote-tracking branch 'refs/remotes/origin/fix/36052-helm-login-host' into fix/36052-helm-login-host
